### ISSUE TITLE
Viash_push exits with code 0 when docker image upload fails. 

### ISSUE THREAD

**Issue by DriesSchaumont** [viash-io/viash#212]
May be releated to #209 

When uploading docker containers using `viash_push` fails, the exit code is still 0.
Example: https://github.com/openpipelines-bio/openpipeline/runs/8109814410?check_suite_focus=true
Check the 'push containers' tab.
<details><summary>Command output</summary>

```
Run bin/viash_push -m release -t integration_build --force
Force push... handle with care...
In RELEASE mode with tag 'integration_build'.
Exporting cellranger_mkfastq (demux) =docker=> target/docker/demux/cellranger_mkfastq
Exporting pca (dimred) =docker=> target/docker/dimred/pca
Notice:  Pushing image to 'ghcr.io/openpipelines-bio/dimred_pca:integration_build'
Notice:  Pushing image to 'ghcr.io/openpipelines-bio/demux_cellranger_mkfastq:integration_build'
Error:  Container 'ghcr.io/openpipelines-bio/dimred_pca:integration_build' push errored. You might not be logged in or have the necessary permissions.
Exporting umap (dimred) =docker=> target/docker/dimred/umap
Error:  Container 'ghcr.io/openpipelines-bio/demux_cellranger_mkfastq:integration_build' push errored. You might not be logged in or have the necessary permissions.
Exporting fastqc (qc) =docker=> target/docker/qc/fastqc
Notice:  Pushing image to 'ghcr.io/openpipelines-bio/dimred_umap:integration_build'
Notice:  Pushing image to 'ghcr.io/openpipelines-bio/qc_fastqc:integration_build'
Error:  Container 'ghcr.io/openpipelines-bio/dimred_umap:integration_build' push errored. You might not be logged in or have the necessary permissions.
Exporting cellranger_count_split (mapping) =docker=> target/docker/mapping/cellranger_count_split
Error:  Container 'ghcr.io/openpipelines-bio/qc_fastqc:integration_build' push errored. You might not be logged in or have the necessary permissions.
Exporting multiqc (qc) =docker=> target/docker/qc/multiqc
Notice:  Pushing image to 'ghcr.io/openpipelines-bio/mapping_cellranger_count_split:integration_build'
Notice:  Pushing image to 'ghcr.io/openpipelines-bio/qc_multiqc:integration_build'
Error:  Container 'ghcr.io/openpipelines-bio/mapping_cellranger_count_split:integration_build' push errored. You might not be logged in or have the necessary permissions.
Exporting cellranger_count (mapping) =docker=> target/docker/mapping/cellranger_count
Error:  Container 'ghcr.io/openpipelines-bio/qc_multiqc:integration_build' push errored. You might not be logged in or have the necessary permissions.
Exporting split_modalities (split) =docker=> target/docker/split/split_modalities
Notice:  Pushing image to 'ghcr.io/openpipelines-bio/mapping_cellranger_count:integration_build'
Error:  Container 'ghcr.io/openpipelines-bio/mapping_cellranger_count:integration_build' push errored. You might not be logged in or have the necessary permissions.
Exporting bd_rhapsody_wta (mapping) =docker=> target/docker/mapping/bd_rhapsody_wta
Notice:  Pushing image to 'ghcr.io/openpipelines-bio/split_split_modalities:integration_build'
Error:  Container 'ghcr.io/openpipelines-bio/split_split_modalities:integration_build' push errored. You might not be logged in or have the necessary permissions.
Exporting log1p (transform) =docker=> target/docker/transform/log1p
Notice:  Pushing image to 'ghcr.io/openpipelines-bio/mapping_bd_rhapsody_wta:integration_build'
Error:  Container 'ghcr.io/openpipelines-bio/mapping_bd_rhapsody_wta:integration_build' push errored. You might not be logged in or have the necessary permissions.
Exporting filter_with_counts (filter) =docker=> target/docker/filter/filter_with_counts
Notice:  Pushing image to 'ghcr.io/openpipelines-bio/transform_log1p:integration_build'
Error:  Container 'ghcr.io/openpipelines-bio/transform_log1p:integration_build' push errored. You might not be logged in or have the necessary permissions.
Exporting scale (transform) =docker=> target/docker/transform/scale
Notice:  Pushing image to 'ghcr.io/openpipelines-bio/filter_filter_with_counts:integration_build'
Error:  Container 'ghcr.io/openpipelines-bio/filter_filter_with_counts:integration_build' push errored. You might not be logged in or have the necessary permissions.
Exporting do_filter (filter) =docker=> target/docker/filter/do_filter
Notice:  Pushing image to 'ghcr.io/openpipelines-bio/transform_scale:integration_build'
Error:  Container 'ghcr.io/openpipelines-bio/transform_scale:integration_build' push errored. You might not be logged in or have the necessary permissions.
Exporting normalize_total (transform) =docker=> target/docker/transform/normalize_total
Notice:  Pushing image to 'ghcr.io/openpipelines-bio/filter_do_filter:integration_build'
Error:  Container 'ghcr.io/openpipelines-bio/filter_do_filter:integration_build' push errored. You might not be logged in or have the necessary permissions.
Exporting filter_with_scrublet (filter) =docker=> target/docker/filter/filter_with_scrublet
Notice:  Pushing image to 'ghcr.io/openpipelines-bio/transform_normalize_total:integration_build'
Error:  Container 'ghcr.io/openpipelines-bio/transform_normalize_total:integration_build' push errored. You might not be logged in or have the necessary permissions.
Exporting regress_out (transform) =docker=> target/docker/transform/regress_out
Notice:  Pushing image to 'ghcr.io/openpipelines-bio/filter_filter_with_scrublet:integration_build'
Error:  Container 'ghcr.io/openpipelines-bio/filter_filter_with_scrublet:integration_build' push errored. You might not be logged in or have the necessary permissions.
Exporting filter_with_hvg (filter) =docker=> target/docker/filter/filter_with_hvg
Notice:  Pushing image to 'ghcr.io/openpipelines-bio/transform_regress_out:integration_build'
Error:  Container 'ghcr.io/openpipelines-bio/transform_regress_out:integration_build' push errored. You might not be logged in or have the necessary permissions.
Exporting harmonypy (integrate) =docker=> target/docker/integrate/harmonypy
Notice:  Pushing image to 'ghcr.io/openpipelines-bio/filter_filter_with_hvg:integration_build'
Error:  Container 'ghcr.io/openpipelines-bio/filter_filter_with_hvg:integration_build' push errored. You might not be logged in or have the necessary permissions.
Exporting from_bdrhap_to_h5mu (convert) =docker=> target/docker/convert/from_bdrhap_to_h5mu
Notice:  Pushing image to 'ghcr.io/openpipelines-bio/integrate_harmonypy:integration_build'
Notice:  Pushing image to 'ghcr.io/openpipelines-bio/convert_from_bdrhap_to_h5mu:integration_build'
Error:  Container 'ghcr.io/openpipelines-bio/convert_from_bdrhap_to_h5mu:integration_build' push errored. You might not be logged in or have the necessary permissions.
Exporting from_csv_to_h5mu (convert) =docker=> target/docker/convert/from_csv_to_h5mu
Notice:  Pushing image to 'ghcr.io/openpipelines-bio/convert_from_csv_to_h5mu:integration_build'
Error:  Container 'ghcr.io/openpipelines-bio/convert_from_csv_to_h5mu:integration_build' push errored. You might not be logged in or have the necessary permissions.
Exporting from_h5ad_to_h5mu (convert) =docker=> target/docker/convert/from_h5ad_to_h5mu
Notice:  Pushing image to 'ghcr.io/openpipelines-bio/convert_from_h5ad_to_h5mu:integration_build'
Error:  Container 'ghcr.io/openpipelines-bio/convert_from_h5ad_to_h5mu:integration_build' push errored. You might not be logged in or have the necessary permissions.
Exporting from_10xmtx_to_h5mu (convert) =docker=> target/docker/convert/from_10xmtx_to_h5mu
Notice:  Pushing image to 'ghcr.io/openpipelines-bio/convert_from_10xmtx_to_h5mu:integration_build'
Error:  Container 'ghcr.io/openpipelines-bio/convert_from_10xmtx_to_h5mu:integration_build' push errored. You might not be logged in or have the necessary permissions.
Exporting from_h5mu_to_seurat (convert) =docker=> target/docker/convert/from_h5mu_to_seurat
Notice:  Pushing image to 'ghcr.io/openpipelines-bio/convert_from_h5mu_to_seurat:integration_build'
Error:  Container 'ghcr.io/openpipelines-bio/convert_from_h5mu_to_seurat:integration_build' push errored. You might not be logged in or have the necessary permissions.
Exporting from_10xh5_to_h5mu (convert) =docker=> target/docker/convert/from_10xh5_to_h5mu
Notice:  Pushing image to 'ghcr.io/openpipelines-bio/convert_from_10xh5_to_h5mu:integration_build'
Error:  Container 'ghcr.io/openpipelines-bio/convert_from_10xh5_to_h5mu:integration_build' push errored. You might not be logged in or have the necessary permissions.
Exporting bcl2fastq (demux) =docker=> target/docker/demux/bcl2fastq
Notice:  Pushing image to 'ghcr.io/openpipelines-bio/demux_bcl2fastq:integration_build'
Error:  Container 'ghcr.io/openpipelines-bio/demux_bcl2fastq:integration_build' push errored. You might not be logged in or have the necessary permissions.
Exporting bcl_convert (demux) =docker=> target/docker/demux/bcl_convert
Notice:  Pushing image to 'ghcr.io/openpipelines-bio/demux_bcl_convert:integration_build'
Error:  Container 'ghcr.io/openpipelines-bio/demux_bcl_convert:integration_build' push errored. You might not be logged in or have the necessary permissions.
Exporting merge (integrate) =docker=> target/docker/integrate/merge
Notice:  Pushing image to 'ghcr.io/openpipelines-bio/integrate_merge:integration_build'
Error:  Container 'ghcr.io/openpipelines-bio/integrate_merge:integration_build' push errored. You might not be logged in or have the necessary permissions.
Exporting publish (transfer) =docker=> target/docker/transfer/publish
Notice:  Pushing image to 'ghcr.io/openpipelines-bio/transfer_publish:integration_build'
Error:  Container 'ghcr.io/openpipelines-bio/transfer_publish:integration_build' push errored. You might not be logged in or have the necessary permissions.
Exporting leiden (cluster) =docker=> target/docker/cluster/leiden
Notice:  Pushing image to 'ghcr.io/openpipelines-bio/cluster_leiden:integration_build'
Error:  Container 'ghcr.io/openpipelines-bio/cluster_leiden:integration_build' push errored. You might not be logged in or have the necessary permissions.
Exporting find_neighbors (neighbors) =docker=> target/docker/neighbors/find_neighbors
Notice:  Pushing image to 'ghcr.io/openpipelines-bio/neighbors_find_neighbors:integration_build'
Error:  Container 'ghcr.io/openpipelines-bio/neighbors_find_neighbors:integration_build' push errored. You might not be logged in or have the necessary permissions.
Exporting run_cellxgene (interactive) =docker=> target/docker/interactive/run_cellxgene
Notice:  Pushing image to 'ghcr.io/openpipelines-bio/interactive_run_cellxgene:integration_build'
Error:  Container 'ghcr.io/openpipelines-bio/interactive_run_cellxgene:integration_build' push errored. You might not be logged in or have the necessary permissions.
Exporting run_cirrocumulus (interactive) =docker=> target/docker/interactive/run_cirrocumulus
Notice:  Pushing image to 'ghcr.io/openpipelines-bio/interactive_run_cirrocumulus:integration_build'
Error:  Container 'ghcr.io/openpipelines-bio/interactive_run_cirrocumulus:integration_build' push errored. You might not be logged in or have the necessary permissions.
Exporting mermaid (report) =docker=> target/docker/report/mermaid
Notice:  Pushing image to 'ghcr.io/openpipelines-bio/report_mermaid:integration_build'
Error:  Container 'ghcr.io/openpipelines-bio/report_mermaid:integration_build' push errored. You might not be logged in or have the necessary permissions.
Exporting filter_10xh5 (process_10xh5) =docker=> target/docker/process_10xh5/filter_10xh5
Notice:  Pushing image to 'ghcr.io/openpipelines-bio/process_10xh5_filter_10xh5:integration_build'
Error:  Container 'ghcr.io/openpipelines-bio/process_10xh5_filter_10xh5:integration_build' push errored. You might not be logged in or have the necessary permissions.
Exporting download_file (download) =docker=> target/docker/download/download_file
Notice:  Pushing image to 'ghcr.io/openpipelines-bio/download_download_file:integration_build'
Error:  Container 'ghcr.io/openpipelines-bio/download_download_file:integration_build' push errored. You might not be logged in or have the necessary permissions.
Exporting sync_test_resources (download) =docker=> target/docker/download/sync_test_resources
Notice:  Pushing image to 'ghcr.io/openpipelines-bio/download_sync_test_resources:integration_build'
Error:  Container 'ghcr.io/openpipelines-bio/download_sync_test_resources:integration_build' push errored. You might not be logged in or have the necessary permissions.
Notice:  Container 'ghcr.io/openpipelines-bio/integrate_harmonypy:integration_build' push succeeded.
Exporting concat (integrate) =docker=> target/docker/integrate/concat
Notice:  Pushing image to 'ghcr.io/openpipelines-bio/integrate_concat:integration_build'
Error:  Container 'ghcr.io/openpipelines-bio/integrate_concat:integration_build' push errored. You might not be logged in or have the necessary permissions.
Not all configs built successfully
  2/39 configs were disabled
  37/39 configs built successfully
```

</details>

It is quite crucial for viash to produce a non-zero exit code in the context of CI, becuase errors like this go by unnoticed.
I think this can be reproduced by attempting an upload and just cutting off internet connection.

**Comment by tverbeiren** [comment]
As an illustration, we start from the following `docker` platform definition:

```yaml
...
platforms:
  - type: docker
    image: dataintuitive/viash:latest
    target_image: wrongrepo/wrongname:wrongtag
```

Running this through `viash` with `--setup` indeed yields the 

```sh
> viash build config.vsh.yaml -p docker --setup push
[notice] Pushing image to 'wrongrepo/wrongname:wrongtag'
[error] Container 'wrongrepo/wrongname:wrongtag' push errored. You might not be logged in or have the necessary permissions.
> echo $?
0
```

When first building the component, it works:

```sh
> <executable> ---setup push
[notice] Pushing image to 'wrongrepo/wrongname:wrongtag'
[error] Container 'wrongrepo/wrongname:wrongtag' push errored. You might not be logged in or have the necessary permissions.
> echo $?
1
```

**Comment by Grifs** [comment]
Fixed in Viash 0.6.3

```
viash build config.vsh.yaml -p docker --setup push
[notice] Pushing image to 'wrongrepo/wrongname:wrongtag'
[error] Container 'wrongrepo/wrongname:wrongtag' push errored. You might not be logged in or have the necessary permissions.
echo $?
1

```
